jlm/util/HashSet.hpp

```diff
@@ -221,6 +221,22 @@ class HashSet
       Insert(item);
   }
 
+  /**
+   * Modifies this HashSet object to contain only elements that are present in itself and \p other.
+   *
+   * @param other A HashSet to intersect with.
+   */
+  void
+  IntersectWith(const HashSet<ItemType> & other)
+  {
+    auto isContained = [&](const ItemType item)
+    {
+      return !other.Contains(item);
+    };
+
+    RemoveWhere(isContained);
```

Comment:
This will get converted to a functional object, with implicit memory allocation. You could change RemoveWhere to use template parameter (and const& qualify the parameter), and then this works without malloc.

Reply:
I am not sure that I fully follow about the RemoveWhere part of your comment. Could you elaborate, please?

Reply:
Proposal:

`template<typename F>
size_t
RemoveWhere(const F& match)
{
  ...
}`

and then if you pass a lambda type it will just use it "as-is" without converting it to a std::function

Reply:
Addressed here: https://github.com/phate/jlm/pull/269